sechub-cli/src/daimler.com/sechub/util/filepathmatcher.go

```diff
@@ -3,91 +3,38 @@ package util
 
 import (
 	"fmt"
-	"path/filepath"
+	"regexp"
 	"strings"
 )
 
-//
-// This method provides ANT like selectors.
+// Filepathmatch - This method provides ANT like selectors.
+// See https://ant.apache.org/manual/dirtasks.html
 // For example: "**/a*.txt" will accept
-//               - "/home/tester/xyz/a1234.txt
-//               - "/root/a1b.txt
+//               - "/home/tester/xyz/a1234.txt"
+//               - "a1b.txt"
 //
 func Filepathmatch(path string, pattern string) (result bool) {
-	/* ------------------------------ */
-	/* Explanation - how it works ... */
-	/* ------------------------------ */
-	// - we separate pattern into different path patterns by path wildcards into elements
-	// - then the last element is inspected to be a file wildcard
-	// - file wildcards are resolved /tried by golang filepath.Match method
-	// - if this is okay we do our special path wildcard matching
-
-
-	/* separate patterns containing path wildcards */
-	subPattern := strings.Split(pattern, "**/")
-//	fmt.Printf("\n%s - seperated into %q", pattern, subPattern)
-	length := len(subPattern)
-	potentialFilePattern := subPattern[length-1]
 
-	/* ------------------------ */
-	/* handle file name pattern */
-	/* ------------------------ */
-	substring := path
-	if strings.Index(potentialFilePattern, "*") != -1 {
-		/* found a file pattern in last entry */
-		/* change length - so slice not changed but last entry no longer handled later */
-		length--
-
-		/* check if file patter is applicable */
-		lastPathIndex := strings.LastIndex(path, "/")
-		fileName := path
-		indexForCut := lastPathIndex
-		if lastPathIndex != -1 {
-			fileName = path[lastPathIndex:]
-		} else {
-			indexForCut = 0
-		}
-		/* we just use the golang integrated filename matcher:*/
-		matches, err := filepath.Match(potentialFilePattern, fileName)
-		if err != nil {
-			fmt.Println(err)
-		}
-		if !matches {
-			return false
-		}
+	// Let's turn the ant style pattern into a regexp:
+	doublestarPatterns := strings.Split(pattern, "**/")
+	//fmt.Printf("\nbefore: %q path: %q", pattern, path)
```

Comment:
I think that `fmt.Printf` can be removed.

Reply:
Yes - it could be removed since it's only a comment.

But - I found similar in the existing code and for the development it was very helpful and convenient, just to uncomment to get the values while testing.
So I just continued that "tradition" because I find it helpful.
I think, it won't harm if we keep the line. :-)

Reply:
Please don't keep dead code inside your git repo. Let us create a "go snippet wiki page" or something to hold such snippets and remove it from the code base. When a commented line is inside code when commit - okay, this can happen. But it should be accidently and removed after recognition.